File: src/Python/ImageData/ImageWeightedSum.py
#!/usr/bin/env python

import vtk


def main():
    colors = vtk.vtkNamedColors()

    colors.SetColor('leftBkg', [0.6, 0.5, 0.4, 1.0])
    colors.SetColor('centreBkg', [0.1, 0.5, 0.4, 1.0])
    colors.SetColor('rightBkg', [0.4, 0.5, 0.6, 1.0])

    # Create image 1
    source1 = vtk.vtkImageMandelbrotSource()
    source1.SetWholeExtent(0, 255, 0, 255, 0, 0)
    source1.Update()

    source1Double = vtk.vtkImageCast()
    source1Double.SetInputConnection(0, source1.GetOutputPort())
    source1Double.SetOutputScalarTypeToDouble()

    # Create image 2
    source2 = vtk.vtkImageSinusoidSource()
    source2.SetWholeExtent(0, 255, 0, 255, 0, 0)
    source2.Update()

    # Do the sum
    sumFilter = vtk.vtkImageWeightedSum()
    sumFilter.SetWeight(0, 0.8)
    sumFilter.SetWeight(1, 0.2)
    sumFilter.AddInputConnection(source1Double.GetOutputPort())
    sumFilter.AddInputConnection(source2.GetOutputPort())
    sumFilter.Update()

    # Display the images
    source1CastFilter = vtk.vtkImageCast()
    source1CastFilter.SetInputConnection(source1.GetOutputPort())
    source1CastFilter.SetOutputScalarTypeToUnsignedChar()
    source1CastFilter.Update()

    source2CastFilter = vtk.vtkImageCast()
    source2CastFilter.SetInputConnection(source2.GetOutputPort())
    source2CastFilter.SetOutputScalarTypeToUnsignedChar()
    source2CastFilter.Update()

    summedCastFilter = vtk.vtkImageCast()
    summedCastFilter.SetInputConnection(sumFilter.GetOutputPort())
    summedCastFilter.SetOutputScalarTypeToUnsignedChar()
    summedCastFilter.Update()

    # Create actors
    source1Actor = vtk.vtkImageActor()
    source1Actor.GetMapper().SetInputConnection(source1CastFilter.GetOutputPort())

    source2Actor = vtk.vtkImageActor()
    source2Actor.GetMapper().SetInputConnection(source2CastFilter.GetOutputPort())

    summedActor = vtk.vtkImageActor()
    summedActor.GetMapper().SetInputConnection(summedCastFilter.GetOutputPort())

    # There will be one render window
    renderWindow = vtk.vtkRenderWindow()
    renderWindow.SetSize(600, 300)

    # And one interactor
    interactor = vtk.vtkRenderWindowInteractor()
    interactor.SetRenderWindow(renderWindow)

    # Define viewport ranges
    # (xmin, ymin, xmax, ymax)
    leftViewport = [0.0, 0.0, 0.33, 1.0]
    centerViewport = [0.33, 0.0, .66, 1.0]
    rightViewport = [0.66, 0.0, 1.0, 1.0]

    # Setup renderers
    leftRenderer = vtk.vtkRenderer()
    renderWindow.AddRenderer(leftRenderer)
    leftRenderer.SetViewport(leftViewport)
    leftRenderer.SetBackground(colors.GetColor3d('leftBkg'))

    centerRenderer = vtk.vtkRenderer()
    renderWindow.AddRenderer(centerRenderer)
    centerRenderer.SetViewport(centerViewport)
    centerRenderer.SetBackground(colors.GetColor3d('centreBkg'))

    rightRenderer = vtk.vtkRenderer()
    renderWindow.AddRenderer(rightRenderer)
    rightRenderer.SetViewport(rightViewport)
    rightRenderer.SetBackground(colors.GetColor3d('rightBkg'))

    leftRenderer.AddActor(source1Actor)
    centerRenderer.AddActor(source2Actor)
    rightRenderer.AddActor(summedActor)

    leftRenderer.ResetCamera()
    centerRenderer.ResetCamera()
    rightRenderer.ResetCamera()

    renderWindow.Render()
    interactor.Start()


if __name__ == '__main__':
    main()

File: src/Java/ImageData/ImageWeightedSum.java
import vtk.vtkNativeLibrary;
import vtk.vtkRenderWindow;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkRenderer;
import vtk.vtkNamedColors;
import vtk.vtkImageCast;
import vtk.vtkImageMandelbrotSource;
import vtk.vtkImageSinusoidSource;
import vtk.vtkImageWeightedSum;
import vtk.vtkImageActor;

public class ImageWeightedSum 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------

  public static void main(String args[]) 
  {
  
    vtkNamedColors Color = new vtkNamedColors(); 
 
    double bkgL[] = new double[4] ;
    double bkgC[] = new double[4] ;
    double bkgR[] = new double [4] ;
         
    Color.GetColor("Tan",bkgL);
    Color.GetColor("DarkCyan",bkgC);
    Color.GetColor("LightSlateGray",bkgR);
    
    //create an image
    vtkImageMandelbrotSource source1 = new vtkImageMandelbrotSource();
    source1.SetWholeExtent(0, 255, 0, 255, 0, 0);
    source1.Update();

    vtkImageCast source1Double = new vtkImageCast();
    source1Double.SetInputConnection(0, source1.GetOutputPort());
    source1Double.SetOutputScalarTypeToDouble();

    vtkImageSinusoidSource source2 = new vtkImageSinusoidSource();
    source2.SetWholeExtent(0, 255, 0, 255, 0, 0);
    source2.Update();
    
    vtkImageWeightedSum sumFilter = new vtkImageWeightedSum();
    sumFilter.SetWeight(0, .8);
    sumFilter.SetWeight(1, .2);
    sumFilter.AddInputConnection(source1Double.GetOutputPort());
    sumFilter.AddInputConnection(source2.GetOutputPort());

    sumFilter.Update();

    vtkImageCast source1CastFilter = new vtkImageCast();
    source1CastFilter.SetInputConnection(source1.GetOutputPort());
    source1CastFilter.SetOutputScalarTypeToUnsignedChar();
    source1CastFilter.Update();

    vtkImageCast source2CastFilter = new vtkImageCast();
    source2CastFilter.SetInputConnection(source2.GetOutputPort());
    source2CastFilter.SetOutputScalarTypeToUnsignedChar();
    source2CastFilter.Update();

    vtkImageCast summedCastFilter = new vtkImageCast();
    summedCastFilter.SetInputConnection(sumFilter.GetOutputPort());
    summedCastFilter.SetOutputScalarTypeToUnsignedChar();
    summedCastFilter.Update();

    //Create actors
    vtkImageActor source1Actor = new vtkImageActor();
    source1Actor.GetMapper().SetInputConnection(source1CastFilter.GetOutputPort());

    vtkImageActor source2Actor = new vtkImageActor();
    source2Actor.GetMapper().SetInputConnection(source2CastFilter.GetOutputPort());

    vtkImageActor summedActor = new vtkImageActor();
    summedActor.GetMapper().SetInputConnection(summedCastFilter.GetOutputPort());
    
    // There will be one render window
    vtkRenderWindow renderWindow = new vtkRenderWindow();
    renderWindow.SetSize(600, 300);

    // And one interactor
    vtkRenderWindowInteractor interactor = new vtkRenderWindowInteractor();
    interactor.SetRenderWindow(renderWindow);

    // Define viewport ranges
    // (xmin, ymin, xmax, ymax)
    double leftViewport[] = new double[] {0.0, 0.0, 0.33, 1.0};
    double centerViewport[] = new double[]{0.33, 0.0, .66, 1.0};
    double rightViewport[] = new double[] {0.66, 0.0, 1.0, 1.0};

    // Setup renderers
    vtkRenderer leftRenderer = new vtkRenderer();
    renderWindow.AddRenderer(leftRenderer);
    leftRenderer.SetViewport(leftViewport);
    leftRenderer.SetBackground(bkgL);

    vtkRenderer centerRenderer = new vtkRenderer();
    renderWindow.AddRenderer(centerRenderer);
    centerRenderer.SetViewport(centerViewport);
    centerRenderer.SetBackground(bkgC);

    vtkRenderer rightRenderer = new vtkRenderer();
    renderWindow.AddRenderer(rightRenderer);
    rightRenderer.SetViewport(rightViewport);
    rightRenderer.SetBackground(bkgR);

    leftRenderer.AddActor(source1Actor);
    centerRenderer.AddActor(source2Actor);
    rightRenderer.AddActor(summedActor);

    leftRenderer.ResetCamera();
    centerRenderer.ResetCamera();
    rightRenderer.ResetCamera();

    renderWindow.Render();
    interactor.Start();
  }
}
